### PR TITLE
Make rectangular FoV marking follow real projection

### LINKED ISSUE #3015: rectangular FOV wrong for other than very small angles
### Expected Behaviour

The angle measurement tool should measure the exact sizes that have been set in the rectangle FOV.

Unless I'm mistaken, this is a known issue.

### Actual Behaviour

The rectangular FOV does not take projection into account. 
(Exception: in cylindrical projection, the vertical dimension is respected)

image

### PR DESCRIPTION
**Pull request by 10110111**
Fixes #3015

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] This change requires a documentation update

### How Has This Been Tested?

**Test Configuration**:
* Operating system: Ubuntu 20.04
* Graphics Card: Intel UHD Graphics 620

## Checklist:

- [x] My code follows the [code style](http://stellarium.org/doc/head/codingStyle.html) of this project.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (header file)
- [ ] I have updated the respective chapter in the Stellarium User Guide
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by alex-w — APPROVED** [review]
Oh, thanks for quick solution!

The fix is not changed translatable strings, so, we can accept it for release 23.2 - is it OK for all?

File: src/core/modules/SpecialMarkersMgr.cpp [hunk 1]
```diff
@@ -131,38 +195,18 @@ void SpecialSkyMarker::draw(StelCore *core) const
...
-			b = transform.map(QPoint(static_cast<int>(-width*0.5), static_cast<int>(height*0.5)));
-			sPainter.drawLine2d(a.x(), a.y(), b.x(), b.y());
-			// right line
-			a = transform.map(QPoint(static_cast<int>(width*0.5), static_cast<int>(height*0.5)));
-			b = transform.map(QPoint(static_cast<int>(width*0.5), static_cast<int>(-height*0.5)));
-			sPainter.drawLine2d(a.x(), a.y(), b.x(), b.y());
-
-			/*
-			 * NOTE: uncomment the code for display FOV value in top right corner of marker
-			if (fovRatio>=0.25)
```
Comment (gzotti) [inline src/core/modules/SpecialMarkersMgr.cpp:159]: @alex-w shall this be permanently deleted?

**Review by gzotti — COMMENTED** [review]
When changing to equatorial (telescope) mode, the box is now tilted. @alex-w is this what you wanted?

File: src/core/modules/SpecialMarkersMgr.cpp (hunk 1 above)
Comment (alex-w) [reply src/core/modules/SpecialMarkersMgr.cpp:159]: It's a "hidden feature" :) As far nobody asked for it before, so, this part may be removed.

**Comment by alex-w** [comment]
> When changing to equatorial (telescope) mode, the box is now tilted. @alex-w is this what you wanted?

This is good question. Previous solution didn't follow the mount types, but the current solution is followed it (`derotate` parameter). I think now it's better.

**Comment by gzotti** [comment]
I am less sure about that and would see using the current mode more intuitive. However, if you like it being always in Azimuthal mode, it's OK to merge. I only use the Ocular plugin's sensor frame.

**Pull request merged**